src/Wsdl/Event/RequestPreEvent.php

```diff
@@ -0,0 +1,123 @@
+<?php
+declare(strict_types = 1);
+
+namespace Skaut\Skautis\Wsdl\Event;
+
+use Serializable;
+
+class RequestPreEvent implements Serializable
+{
+
+    /**
+     * @var string Nazev funkce volane pomoci SOAP requestu
+     */
+    private $fname;
+
+    /**
+     * Parametry SOAP requestu na server
+     *
+     * @var array<int, mixed>
+     */
+    private $args;
+
+    /**
+     * @var array<string, mixed>
+     */
+    private $options;
+
+    /**
+     * @var array<int, string>
+     */
+    private $inputHeaders;
+
+    /**
+     * @var array<int, array<string, mixed>> Zasobnik volanych funkci
+     */
+    private $trace;
+
+
+    /**
+     * @param string $fname Nazev volane funkce
+     * @param array<int, mixed> $args  Argumenty pozadavku
+     * @param array<string, mixed> $options
+     * @param array<int, string> $inputHeaders
+     * @param array<int, array<string, mixed>> $trace Zasobnik volanych funkci
+     */
+    public function __construct(
+      string $fname,
+      array &$args,
+      array &$options,
+      array &$inputHeaders,
+      array $trace
+    ) {
+        $this->fname = $fname;
+        $this->args = &$args;
+        $this->options = &$options;
+        $this->inputHeaders = &$inputHeaders;
+        $this->trace = $trace;
+    }
+
+    public function serialize(): string
+    {
+        $data = [
+            'fname' => $this->fname,
+            'args' => $this->args,
+            'options' => $this->options,
+            'inputHeaders' => $this->inputHeaders,
+            'trace' => $this->trace,
+        ];
+        return serialize($data);
+    }
+
+    /**
+     * @param string $data
+     */
+    public function unserialize($data): void
+    {
+        $data = unserialize($data, [self::class]);
+        $this->fname = (string) $data['fname'];
+        $this->args = (array) $data['args'];
+        $this->options = (array) $data['options'];
+        $this->inputHeaders = (array) $data['inputHeaders'];
+        $this->trace = (array) $data['trace'];
+    }
+
+
+    public function getFname(): string
+    {
+      return $this->fname;
+    }
+
+    /**
+     * @return array<int, mixed>
+     */
+    public function &getArgs(): array
```

Comment:
@JindrichPilar Těm referencím bych se radši vyvaroval. K čemu mají být? Pokud to má být způsob, jak upravit v subscriberu ten daný event, tak je to IMO dost messy řešení. Mnohem lepší je v tu chvíli použít vlastní dekorátor kolem servisy (tím, že dokonce nabízíme i `AbstractDecorator` je to dost snadný způsob), případně vlastní implementací webservice. Tohle totiž může vést k nehezkým bugům.

Reply:
> použít vlastní dekorátor kolem servisy
Ten se nedostane k `options` a  `inputHeaders`.  Bez hacku s reflection.

> Tohle totiž může vést k nehezkým bugům.
Bylo to myšleno jako expert only feature s velkým varováním v dokumentaci.

Vzhledem k #97 jsem to ale odstranil, nejsem si jistý jak moc by to bylo kompatibilní.